lib/internal/modules/esm/loader.js

```diff
@@ -155,30 +155,33 @@ class ESMLoader {
    *                                                  module import.
    * @returns {Promise<ModuleJob>} The (possibly pending) module job
    */
-  async getModuleJob(specifier, parentURL, importAssertions) {
+  async getModuleJob(specifier, parentURL, importAssertions = { __proto__: null }) {
```

Comment:
```suggestion
  async getModuleJob(specifier, parentURL, importAssertions = kEmptyObject) {
```

Reply:
Should this be optional in fact? Maybe it's better to keep it as required, wdyt?
```suggestion
  async getModuleJob(specifier, parentURL, importAssertions) {
```

Reply:
`context.importAssertions` is a copy, not the original, so it wouldn't be frozen.

Reply:
> Should this be optional in fact? Maybe it’s better to keep it as required, wdyt?

It was optional before this PR, but the code in this method took for granted that it was always defined (because the places that call this method always define it). I didn’t want to rely on that, though, as it felt brittle.

Reply:
How do you mean it was optional?

Reply:
> How do you mean it was optional?

Before this PR it had no default value. It could have been undefined, but it never was because of the `{ __proto__: null }` in the default value in the loader’s `resolve` method.

Reply:
Yeah that's every value in JS though, what's brittle about it?

Reply:
What’s brittle is that `getModuleJob` reads its `type` property without first checking that it could be undefined. So either I should add a guard like `?.type` or the default value, to avoid a potential crash on trying to read a property of an undefined object if a future refactor ever makes it possible to call `getModuleJob` with an undefined `importAssertions` parameter.

Reply:
The signature in `resolve` is `resolve(originalSpecifier, parentURL, importAssertions = { __proto__: null })`, I kind of feel like we should just copy that for consistency. If we changed both to `importAssertions = kEmptyObject` I’d have to add cloning to `resolve` which is unnecessary.

Reply:
I've opened https://github.com/nodejs/node/pull/46164 to address this. I'd say we want a crash if `undefined` is ever supplied to that _internal_ function.

Reply:
Okay, can you just add that PR to this one? I don’t want to wait 2 days to land that and then rebase this on top of it and wait another 2 days.

Reply:
You don't have to wait, you could keep the parameter as required in this PR (i.e. remove the default value you added), they shouldn't conflict with one another, they touch different lines.